### ISSUE TITLE
pgpool.connect() raises multple errors on failure

### ISSUE THREAD

**Issue by irakliy81** [brianc/node-postgres#1170]
When `pgpool.connect()` fails (e.g. database name was supplied incorrectly), the error is raised in 4 different ways, and if the app is not handling all of them, the process crushes.

The 4 ways are:
1. Error is returned as the first parameter in the callback
2. `pgpool.connect()` returns a rejected promise. I would think that if the callback is provided, there is no need to return a rejected promise
3. pgpool's `error` event is emitted with the same error as in the callback and rejected promise. I would think that if the error was handled in the callback, for example, there is no need to emit it from pgpool
4. pgpool's `error` event is emitted again, with a different error (client closing) - this is somewhat confusing because the client creation failed

Overall, I would expect that if the error is handled in the callback the other 3 errors should not be raised.

**Comment by charmander** [comment]
I think these were fixed by brianc/node-pg-pool#31.

**Comment by charmander** [comment]
I’m not seeing an error event emitted here:

```javascript
'use strict';

var pg = require('pg');

var pool = new pg.Pool({
	database: 'missing',
});

pool.connect()
	.then(
		client => {
			console.log('success');
			client.release();
		},
		() => {
			console.log('failure');
		}
	)
	.then(
		() => { pool.end(); }
	);
```

– just the expected promise rejection. Could you provide a small example that produces (4), please?
